google-cloud-logging/src/test/java/com/google/cloud/logging/TimestampDefaultFilterTest.java

```diff
@@ -33,23 +31,20 @@ public class TimestampDefaultFilterTest {
   public void DefaultTimestampFilterTest() {
     ITimestampDefaultFilter filter = new TimestampDefaultFilter();
 
-    TimeZone timeZone = TimeZone.getTimeZone("UTC");
-    Calendar calendar = Calendar.getInstance(timeZone);
-    calendar.add(Calendar.DATE, -1);
-    Date expected = calendar.getTime();
-
     // Timestamp filter exists
     String defaultFilter = filter.createDefaultTimestampFilter();
-    assertTrue(defaultFilter.contains("timestamp>="));
+    assertThat(defaultFilter).contains("timestamp>=");
 
     // Time is last 24 hours
-    try {
-      DateFormat rfcDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-      rfcDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-      Date actual = rfcDateFormat.parse(defaultFilter.substring(12, defaultFilter.length() - 1));
-      assertTrue(Math.abs(expected.getTime() - actual.getTime()) < Timer.ONE_MINUTE);
-    } catch (java.text.ParseException ex) {
-      fail(); // Just fail if exception is thrown
-    }
+    DateTimeFormatter rfcDateFormat =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", US);
```

Comment:
Should it be UTC?

Reply:
[This is Locale, not timezone](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ofPattern-java.lang.String-). Locale would be used if the pattern had something Locale-specific, like a month name. AFAICT there aren't any Locale-dependent components in this parse string so it's kind of a no-op, but nonetheless our linter encourages us to always specify local because otherwise it's read from the environment, which makes the system unnecessarily environment-dependent. (This timestamp is being generated as a Cloud Logging filter so it should always use the same Locale, US, to the extent that it matters.)

Meanwhile timezone is still actually being read from the input here; it's the `Z` part of the parse format string.